src/js/modules/RoutingApp.js

```diff
@@ -25,195 +26,143 @@ import {
-                />
-                <Route
-                  exact
-                  path={"/illustrators/:illustrator_slug"}
-                  key={"illustrators"}
-                  render={props => <IllustratorPage match={props.match} />}
-                />
-                <Route
-                  exact
-                  path={"/photographers/:photographer_slug"}
-                  key={"photographers"}
-                  render={props => <PhotographerPage match={props.match} />}
-                />
-                <Route
-                  exact
-                  path={"/myaccount"}
-                  key={"signIn"}
-                  render={() =>
-                    session ? (
-                      <Redirect to="/myaccount/profile" />
-                    ) : (
-                      <SignInPage />
-                    )}
-                />
+const RoutingApp = ({
+  data: { loading, error, allSections },
+  session,
+  descriptions,
+}) => {
+  return (
+    <ConnectedRouter
+      onUpdate={() => window.scrollTo(0, 0)}
+      history={appHistory}
+    >
+      <PageLayout>
+        {!loading ? (
+          <Switch>
+            <Route exact path="/" component={HomePage} />
+            {allSections.map(section => {
+              return (
                 <Route
                   exact
-                  path="/myaccount/sign-up"
-                  key={"signUp"}
-                  render={() =>
-                    session ? (
-                      <Redirect to="/myaccount/profile" />
-                    ) : (
-                      <SignUpPage />
-                    )}
+                  path={section.permalink}
+                  key={`section${section.id}`}
+                  render={props => <SectionPage section={section} />}
                 />
+              );
+            })}
+            {allSections.map(section => {
+              return (
                 <Route
                   exact
-                  path="/myaccount/profile"
-                  key={"profile"}
-                  render={() =>
-                    session ? <ProfilePage /> : <Redirect to="/myaccount" />}
+                  path={`${section.permalink}/:article_slug`}
+                  key={`article${section.id}`}
+                  component={ArticlePage}
                 />
+              );
+            })}
+            {descriptions.map(description => {
+              return (
                 <Route
                   exact
-                  path="/myaccount/profile/edit"
-                  key={"editProfile"}
-                  render={() =>
-                    session ? (
-                      <EditProfilePage />
-                    ) : (
-                      <Redirect to="/myaccount" />
-                    )}
+                  path={`/about/${description.slug}`}
+                  key={`description${description.id}`}
+                  render={props => (
+                    <DescriptionPage description={description} />

```

Comment:
Why not just have the description page fetch the data? 
You should just have a route
`<Route exact path={`/about/:description_slug`} />`
and have DescriptionPage fetch the actual data